### ISSUE TITLE
Hints on changing menu font size in Ctrlr build

### ISSUE THREAD

**Issue by shooking** [RomanKubiak/ctrlr#105]
So I built on Raspberry Pi4. The menu fonts and LUA Editor are huge.
On a 32" 1080p it feels like 640 x 480 :-(

I couldnt find the display font controls - any hints gratefully appreciated.
I have build on Windows before and fonts are good size (I guess under control of Windows font manager).
I did try messing with Xterm size before launching - no difference.

You can see what I mean in one of my [YouTube videos](https://www.youtube.com/watch?v=XWmQfK9QzJk) about 1:50 in.
Huge menu bars etc. Would be great to know how to reduce this please.

**Comment by shooking** [comment]
So this isnt specific to this build - more "in general" how to be able, in a Linux type build, influence the rendered menu font sizes please.

**Comment by RomanKubiak** [comment]
i'm working on making the ctrlr interface more scalable, so that fonts icons and other components scale with the size of the program, but this might take a while

**Comment by shooking** [comment]
Great. Any hints on what I might hack in the meantime. I can see from today
that if I set a font in the Lua editor this is respected for code but not
necessarily the file explorer. And not found control over the Modulator
properties fonts.

But I don't know where to find control for the menu fonts.

On Windows 4k panel and 17" I am grateful for large menu.
On Raspberry Pi 1080p and 32" panel the large font is driving me bonkers.

Ideally I want to use the Pi with 21" touchscreen - I am hoping it will be
a decent basis for controlling a ton of midi stuff in my garage.

Thanks again for all your Ctrlr work. Appreciate  it.

Best regards
Steve

**Comment by RomanKubiak** [comment]
i thought maybe i could implement a global zoom for ctrlr not only for the panel itself. changing font sizes in all components is a lot of code, code i don't want to write cause it's error prone

**Comment by shooking** [comment]
OK so great news is the Pi can pretty much build the latest code with only a few tweaks.
What I found, and this is just for my own benefit, is the following hacks get me further with the fonts

pi@raspberrypi:~/Software/ctrlr/Source $ grep -RA2 "SHOOKING" *
Core/CtrlrManager/CtrlrManager.cpp:	// SHOOKING HACK
Core/CtrlrManager/CtrlrManager.cpp-	// setProperty (Ids::ctrlrFontSizeBaseValue, 14.0f);
Core/CtrlrManager/CtrlrManager.cpp-	setProperty (Ids::ctrlrFontSizeBaseValue, 7.0f);
Core/CtrlrManager/CtrlrManager.cpp:        // SHOOKING HACK
Core/CtrlrManager/CtrlrManager.cpp-	// setProperty (Ids::ctrlrScrollbarThickness, 18.0f);
Core/CtrlrManager/CtrlrManager.cpp-    setProperty (Ids::ctrlrScrollbarThickness, 9.0f);
--
Core/CtrlrManager/CtrlrManager.cpp:    // SHOOKING HACK
Core/CtrlrManager/CtrlrManager.cpp-    // setProperty (Ids::ctrlrMenuBarHeight, 24);
Core/CtrlrManager/CtrlrManager.cpp-    // setProperty (Ids::ctrlrTabBarDepth, 24.0);
Core/CtrlrManager/CtrlrManager.cpp-    setProperty (Ids::ctrlrMenuBarHeight, 12);
Core/CtrlrManager/CtrlrManager.cpp-    setProperty (Ids::ctrlrTabBarDepth, 12.0);

Now 24 point in itself is not so bad -  but on the Pi4 1080p and 32" panel the height seems 2x normal size (approx).

Can you think of any "scale" factors that might be being applied here? In Juce say?

In the meantime for my own benefit only I will add some env var in that I can use to scale 50% - that ought to do it.

Thanks again - on Pi and latest source one needs -latomic in the Juce stuff and some issue with too old a version of binutils ...
Misc/include/libr.h:/* SHOOKING HACK */
Misc/include/libr.h-/* #define HAVE_BFD_2_34 1 */

Not sure how to close this "issue" - it was more of a consult. Thanks again - appreciate it.

**Comment by RomanKubiak** [comment]
There are some discussions on the JUCE forums about how it works on the pi and if they are looking into this. I wish they did more, multi touch support would nice on linux it's still not there. The scaling stuff is your personal settings and that's good, i ran ctrlr on the PI too and i had to tweak those vars too, the BFD stuff should be fixed now at least in a sense that it's clear what to do in case of an error

**Comment by shooking** [comment]
Agreed. Thanks again for support.

**Comment by shooking** [comment]
Ok so it is "solved" - I bought an Iiyama monitor at Christmas - only just unpacked and using it.
This 1k panel feels like a 4k with Ctrlr because all the fonts are rendered decent size.
Infact, now I have to enlarge them!
I made a video about the monitor on YouTube - the original video, captured off the 32" Samsung s530 TV operating as a 1k monitor, takes up whole screen. The same panel on the 22" Iiyama T2252MSC renders the panel at 1/4 screen size. This makes sense as that's the size of the canvas.

My guess is the xdpyinfo is just very wrong with the TV, and very right with the touchscreen.
Added bonus - the touchscreen operates nicely with Ctrlr.

Thanks for help with the issue. Closing as ok.
